### PR TITLE
feat(syn-solidity): improve variable getters generation

### LINKED ISSUE #253: [Feature] Generate getters for public mapping variables
### Component

sol! macro

### Describe the feature you would like

Currently this code:

https://github.com/alloy-rs/core/blob/5267aae3fe588e6c023032a622edbe7d86693341/crates/syn-solidity/src/item/function.rs#L101-L103

Only handles the basic case of `uint256 public x = ...;` -> `function x() returns (uint256)`

This can be expanded to support:
`mapping(type1 key1 => mapping(type2 key2 => ... valueType value)) public map = ...;`
-> `function map( $($type $key),* ) returns (valueType value)`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Closes #253

## Solution

Recurse into mappings in value positions, adding keys to the arguments list, and setting the last non-mapping value as the return value.

## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [x] Breaking changes

### CODE REVIEW

File: crates/syn-solidity/src/attribute/function.rs [hunk 1]
```diff
@@ -155,7 +148,6 @@ impl fmt::Debug for FunctionAttribute {
             Self::Visibility(visibility) => f.debug_tuple("Visibility").field(visibility).finish(),
             Self::Mutability(mutability) => f.debug_tuple("Mutability").field(mutability).finish(),
             Self::Virtual(_) => f.write_str("Virtual"),
-            Self::Immutable(_) => f.write_str("immutable"),
```
Comment (prestwich) [inline crates/syn-solidity/src/attribute/function.rs:158]: drve-by nit: uncaptialize `"Virtual"` on the line above

File: crates/syn-solidity/src/attribute/function.rs [hunk 2]
```diff
@@ -207,11 +197,15 @@ impl Parse for FunctionAttribute {
 }
 
 impl From<VariableAttribute> for FunctionAttribute {
+    /// Converts a variable attribute to its corresponding function attribute.
+    ///
+    /// - `constant` -> `pure`
+    /// - `immutable` -> `view`
```
Comment (prestwich) [inline crates/syn-solidity/src/attribute/function.rs:203]: both `constant` and `immutable` correspond to code regions and `codecopy` commands, no? shoudn't they be the same?

File: crates/syn-solidity/src/attribute/function.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/syn-solidity/src/attribute/function.rs:158]: It's debug fmt

File: crates/syn-solidity/src/attribute/function.rs (hunk 2 above)
Comment (DaniPopes) [reply crates/syn-solidity/src/attribute/function.rs:203]: Doesn't really matter, just needed a way to differentiate the two

**Review by prestwich — APPROVED** [review]

**Pull request merged**